src/Common/Commands.ResourceManager.Common/RMProfileClient.cs

```diff
@@ -233,6 +246,69 @@ private List<AzureTenant> ListAccountTenants(AzureAccount account, AzureEnvironm
             }
         }
 
+        /// <summary>
+        /// List all tenants for the account in the profile context
+        /// </summary>
+        /// <returns>The list of tenants for the default account.</returns>
+        public IEnumerable<AzureTenant> ListTenants()
+        {
+            return ListAccountTenants(_profile.Context.Account, _profile.Context.Environment, null, ShowDialog.Never);
+        }
+
+        private IEnumerable<AzureSubscription> ListSubscriptionsForTenant(AzureAccount account, AzureEnvironment environment, 
```

Comment:
Refactoring: Since TryGetTenantSubscription does List() internally as well, we should consider calling ListSubscriptionsForTenant() inside TryGetTenantSubscription.